src/foremast/datapipeline/datapipeline.py

```diff
@@ -0,0 +1,114 @@
+
+import awscli.customizations.datapipeline.translator as translator
+import boto3
+
+from ..exceptions import ErrorCreatingDataPipeline
+from ..utils import get_details, get_properties
+
+LOG = logging.getLogger(__name__)
+
+
+class AWSDataPipeline(object):
+    """Manipulate Data Pipeline."""
+
+    def __init__(self, app=None, env=None, region='us-east-1', prop_path=None):
+        """AWS Data Pipeline object.
+
+        Args:
+            app (str): Application name
+            env (str): Environment/Account
+            region (str): AWS Region
+            prop_path (str): Path of environment property file
+        """
+        self.app_name = app
+        self.env = env
+        self.region = region
+        self.properties = get_properties(prop_path)
+        self.datapipeline_data = self.properties[self.env]['datapipeline']
+        generated = get_details(app=self.app_name)
+        self.group = generated.data['project']
+
+        session = boto3.Session(profile_name=self.env, region_name=self.region)
+        self.client = session.client('datapipeline')
+        self.pipeline_id = None
+
+    def create_datapipeline(self):
+        """Creates the data pipeline if it does not already exist"""
+
+        tags = [
+                {"key": "app_group",
+                "value": self.group},
+                {"key": "app_name",
+                "value": self.app_name}
+                ]
+        try:
+            response = self.client.create_pipeline(name=self.datapipeline_data.get('name', self.app_name),
+                                               uniqueId=self.app_name,
+                                               description=self.datapipeline_data['description'],
+                                               tags=tags)
+            self.pipeline_id = response.get('pipelineId')
+        except Exception as e:
+            LOG.warning(e)
+            raise ErrorCreatingDataPipeline
+        
+        LOG.debug(response)
+        LOG.info("Successfully configured Data Pipeline - %s", self.app_name)
+
+    def set_pipeline_definition(self):
+        """Translates the json definition and puts it on created pipeline
+
+        Returns:
+                bool: True if successful
+        """
+
+        if not self.pipeline_id:
+            self.get_pipeline_id()
+
+        json_def = self.datapipeline_data['json_definition']
+        try:
+            pipelineObjects = translator.definition_to_api_objects(json_def)
+            parameterObjects = translator.definition_to_api_parameters(json_def)
+            parameterValues = translator.definition_to_parameter_values(json_def)
+            response = self.client.put_pipeline_definition(
+                                pipelineId=self.pipeline_id,
+                                pipelineObjects=pipelineObjects,
+                                parameterObjects=parameterObjects,
+                                parameterValues=parameterValues)
+        except Exception as e:
```

Comment:
Don't catch a broad `Exception`. You were handling `awscli.customizations.datapipeline.translator.PipelineDefinitionError` before, so only handle that scenario. Otherwise, we will end up with masked failures.